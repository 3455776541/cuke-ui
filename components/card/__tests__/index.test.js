import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Card from "../index";
import Button from "../../button";

describe("<Card/>", () => {
  it("should render a <Card/> components", () => {
    const wrapper = render(
      <Card title="黄瓜 ui" extra={<Button href="#">更多</Button>}>
        <div> 内容 1</div>
        <div> 内容 1</div>
        <div> 内容 1</div>
      </Card>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-card classnames", () => {
    const wrapper = shallow(
      <Card title="黄瓜 ui" extra={<Button href="#">更多</Button>}>
        <div> 内容 1</div>
        <div> 内容 1</div>
        <div> 内容 1</div>
      </Card>
    );
    assert(wrapper.find(".cuke-card").length >= 1);
    assert(wrapper.find(".cuke-card-header").length >= 1);
    assert(wrapper.find(".cuke-card-header-title").length >= 1);
    assert(wrapper.find(".cuke-card-header-extra").length >= 1);
    assert(wrapper.find(".cuke-card-content").length >= 1);
  });

  it("should render custom shallow", () => {
    const wrapper = shallow(
      <Card title="黄瓜 ui" extra={<Button href="#">更多</Button>}>
        <div> 内容 1</div>
        <div> 内容 1</div>
        <div> 内容 1</div>
      </Card>
    );
    assert(wrapper.find(".cuke-card-shadow").length === 1);
    wrapper.setProps({
      shadow: false
    });
    assert(wrapper.find(".cuke-card-shadow").length === 0);
  });

  it("should render custom title overflow hidden", () => {
    const wrapper = shallow(
      <Card
        title="黄瓜 ui"
        extra={<Button href="#">更多</Button>}
        titleOverflowHidden={true}
      >
        <div> 内容 1</div>
        <div> 内容 1</div>
        <div> 内容 1</div>
      </Card>
    );
    assert(wrapper.find(".cuke-card-overflow-hidden").length === 1);
    wrapper.setProps({
      titleOverflowHidden: false
    });
    assert(wrapper.find(".cuke-card-overflow-hidden").length === 0);
  });

  it("should can not render title", () => {
    const wrapper = shallow(
      <Card extra={<Button href="#">更多</Button>}>
        <div> 内容 1</div>
        <div> 内容 1</div>
        <div> 内容 1</div>
      </Card>
    );
    assert(wrapper.find(".cuke-card-header-hidden").length === 0);
  });

  it("should can not render title", () => {
    const wrapper = shallow(
      <Card>
        <div> 内容 1</div>
        <div> 内容 1</div>
        <div> 内容 1</div>
      </Card>
    );
    assert(wrapper.find(".cuke-card-header-extra").length === 0);
  });

  it("should find custom extra", () => {
    const wrapper = shallow(
      <Card extra={<Button href="#">更多</Button>}>
        <div> 内容 1</div>
        <div> 内容 1</div>
        <div> 内容 1</div>
      </Card>
    );
    assert(wrapper.find(Button).length === 1);
  });

  it("should find custom content", () => {
    const wrapper = render(
      <Card title="标题" extra={<Button href="#">更多</Button>}>
        <div>内容1</div>
      </Card>
    );
    expect(wrapper.text()).toContain("标题更多内容1");
  });
});
