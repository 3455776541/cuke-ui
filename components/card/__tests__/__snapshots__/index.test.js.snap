// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> should render a <Card/> components 1`] = `
<section
  class="cuke-card cuke-card-shadow"
>
  <div
    class="cuke-card-header"
  >
    <div
      class="cuke-card-header-title cuke-card-overflow-hidden"
    >
      黄瓜 ui
    </div>
    <div
      class="cuke-card-header-extra"
    >
      <a
        class="cuke-button-link"
        disabled=""
        href="#"
      >
        <span>
          更多
        </span>
      </a>
    </div>
  </div>
  <div
    class="cuke-card-content"
  >
    <div>
       内容 1
    </div>
    <div>
       内容 1
    </div>
    <div>
       内容 1
    </div>
  </div>
</section>
`;
